sdk/rust/oak/src/lib.rs

```diff
@@ -593,10 +609,33 @@ macro_rules! entrypoint_command_handler {
         ::oak::entrypoint!($name < _ > => |receiver: ::oak::io::Receiver<_>| {
             use ::oak::io::ReceiverExt;
             ::oak::logger::init_default();
-            ::oak::run_command_loop(<$handler>::default(), receiver.iter());
+            let instance = <$handler>::default();
+            ::oak::run_command_loop(instance, receiver.iter());
+        });
+        impl ::oak::WasmEntrypoint for $handler {
+            const ENTRYPOINT_NAME: &'static str = std::stringify!($name);
+            type Message = <$handler as ::oak::CommandHandler>::Command;
+        }
+    };
+}
+
+/// Similar to [`entrypoint_command_handler`], but for nodes that expect to receive initialization
+/// data via an [`oak_io::InitWrapper`] instance over the inbound channel before processing
+/// subsequent commands.
+#[macro_export]
+macro_rules! entrypoint_command_handler_init {
```

Comment:
I think it can replace `entrypoint_command_handler` and just check whether Init message is `()` and thus don't send it

Reply:
Thanks, that's an interesting idea! basically if init is () then it would mean that the channel is actually a receiver of command instances, right? Not sure it's possible to express this as a single type though, nor how to make the macro match on it. I'll leave it out for now, but please feel free to try it out in a separate PR!